app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt

```diff
@@ -230,4 +236,32 @@ class ConversationViewModel @Inject constructor(
             )
         }
     }
+
+    private suspend fun fromMessageModelToMessageContent(message: Message): MessageContent =
+        when (val content = message.content) {
+            is Asset -> {
+                val assetId = content.value.remoteData.assetId
+                val (imgWidth, imgHeight) = when (val metadata = content.value.metadata) {
+                    is Image -> metadata.width to metadata.height
+                    else -> 0 to 0
+                }
+                val decodedImgDataResult = getMessageAsset(
+                    conversationId = message.conversationId,
+                    messageId = message.id
+                ).run {
+                    when (this) {
+                        is MessageAssetResult.Success -> decodedAsset
+                        else -> null
+                    }
+                }
+                when {
+                    content.value.mimeType.contains("image") -> ImageMessage(decodedImgDataResult, width = imgWidth, height = imgHeight)
+                    assetId.isEmpty() -> TextMessage(messageBody = MessageBody("The asset message could not be downloaded correctly"))
```

Comment:
can we add those strings to the strings.xml file

Reply:
This is not final. We need to define an error UI to handle this, so I would avoid to include it to the strings file so that it doesn't create unneeded translations. I added a TODO comment instead